django_projects/locallibrary/catalog/views.py

```diff
@@ -1,3 +1,25 @@
 from django.shortcuts import render
+from catalog.models import Book, Author, BookInstance, Genre
 
-# Create your views here.
+def index(request):
+    """View function for home page of site"""
+
+    # Generate counts of the main objects
+    num_books = Book.objects.count()
+    num_instances = BookInstance.objects.count()
+
+    # Available books (status = 'a')
+    num_instances_available = BookInstance.objects.filter(status__exact='a').count()
```

Comment:
Tìm hiểu các Fields lookup khác ngoài `exact`

Reply:
Một số field lookups:
exact: đúng giá trị đang tìm. Như trường hợp này là đúng chữ cái 'a'
iexact: đúng giá trị đang tìm nhưng có thể thay đổi chữ viết hoa, viết thường. Trong trường hợp này sẽ là chữ cái 'a' hoặc 'A'
endswith và startswith: kết thúc hoặc bắt đầu với giá trị. Trong trường hợp này là kết thúc hoặc bắt đầu với chữ 'a'
iendswith và istartswith: kết thúc hoặc bắt đầu với giá trị. Trong trường hợp này là kết thúc hoặc bắt đầu bằng 'a' hoặc 'A'
gt hoặc lt: lớn hơn hoặc nhỏ hơn giá trị
in: khớp với một trong những giá trị trong list
isNull: giá trị NULL
date: trùng với ngày
day, week, month, year: trùng với ngày, tuần, tháng, năm